### PR TITLE
Fix upgrade not working

### LINKED ISSUE #1216: nativefier --upgrade saves upgraded app in current working directory instead of directory where app currently is located
**Homework**

- [x] I looked at `nativefier --help` and https://github.com/nativefier/nativefier/blob/master/API.md
- [x] I checked the [Troubleshooting section of the README](https://github.com/nativefier/nativefier/blob/master/README.md#troubleshooting).
- [ ] I searched existing issues, open & closed. Yes, my bug is new.
- [x] I'm using the latest version available at https://github.com/nativefier/nativefier/releases

**Bug description**

`nativefier --upgrade` saves upgraded app in current working directory instead of directory where app currently is located. 

**Steps to reproduce**

```
❯ nativefier --upgrade ~/Applications/Example\ Domain-darwin-x64/Example\ Domain.app --verbose
Running in verbose mode! This will produce a mountain of logs and is recommended only for troubleshooting or if you like Shakespeare.

Performing async options post-processing.
mozilla/5.0 (macintosh; intel mac os x 10_12_5) applewebkit/537.36 (khtml, like gecko) chrome/89.0.4389.82 safari/537.36 not found in [ 'edge', 'firefox', 'safari' ]
Got icon from options. Using it, no inferring needed
Got name Example Domain from options. No inferring needed
Sanitized filename for Example Domain : Example Domain
options {
  packager: {
    appCopyright: undefined,
    appVersion: undefined,
    arch: 'x64',
    asar: false,
    buildVersion: undefined,
    darwinDarkModeSupport: true,
    dir: '/Users/USER/.n/lib/node_modules/nativefier/app',
    electronVersion: '12.0.10',
    icon: '/Users/USER/Applications/Example Domain-darwin-x64/Example Domain.app/Contents/Resources/electron.icns',
    name: 'Example Domain',
    out: '/Users/USER',
    overwrite: true,
    platform: 'darwin',
    portable: false,
    targetUrl: 'https://example.com/',
    tmpdir: false,
    upgrade: true,
    upgradeFrom: '/Users/USER/Applications/Example Domain-darwin-x64/Example Domain.app',
    win32metadata: { ProductName: '', InternalName: '', FileDescription: '' }
  },
  nativefier: {
    accessibilityPrompt: true,
    alwaysOnTop: false,
    backgroundColor: null,
    basicAuthPassword: null,
    basicAuthUsername: null,
    blockExternalUrls: false,
    bookmarksMenu: null,
    bounce: false,
    browserwindowOptions: {},
    clearCache: false,
    counter: false,
    crashReporter: undefined,
    disableContextMenu: false,
    disableDevTools: false,
    disableGpu: false,
    diskCacheSize: null,
    disableOldBuildWarning: false,
    enableEs3Apis: false,
    fastQuit: false,
    fileDownloadOptions: {},
    flashPluginDir: null,
    fullScreen: false,
    globalShortcuts: null,
    hideWindowFrame: false,
    ignoreCertificate: false,
    ignoreGpuBlacklist: false,
    inject: [],
    insecure: false,
    internalUrls: null,
    lang: undefined,
    maximize: false,
    nativefierVersion: '44.0.0',
    processEnvs: {},
    proxyRules: null,
    showMenuBar: false,
    singleInstance: false,
    titleBarStyle: null,
    tray: false,
    userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36',
    userAgentHonest: false,
    verbose: true,
    versionString: undefined,
    width: 1280,
    height: 800,
    minWidth: undefined,
    minHeight: undefined,
    maxWidth: undefined,
    maxHeight: undefined,
    widevine: false,
    x: undefined,
    y: undefined,
    zoom: 1
  }
}

Preparing Electron app...
Copying electron app from /Users/USER/.n/lib/node_modules/nativefier/app to /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-16-52-34-app--38131-223qXQkZDU4B
Writing app config to /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-16-52-34-app--38131-223qXQkZDU4B/nativefier.json
No files to inject, skipping copy.
Updating /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-16-52-34-app--38131-223qXQkZDU4B/package.json 'name' field to example-domain-nativefier-182cce

Converting icons...
Building for macOS and icon is already a .icns, no conversion needed
Copying icons if necessary
No copying necessary on macOS; aborting

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn't cached yet...
  electron-packager Electron Packager 15.2.0
  electron-packager Node v16.3.0
  electron-packager Host Operating system: darwin 17.7.0 (x64) +0ms
  electron-packager Packager Options: {"arch":"x64","asar":false,"darwinDarkModeSupport":true,"dir":"/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-16-52-34-app--38131-223qXQkZDU4B","electronVersion":"12.0.10","icon":"/Users/USER/Applications/Example Domain-darwin-x64/Example Domain.app/Contents/Resources/electron.icns","name":"Example Domain","out":"/Users/USER","overwrite":true,"platform":"darwin","portable":false,"targetUrl":"https://example.com/","tmpdir":false,"upgrade":true,"upgradeFrom":"/Users/USER/Applications/Example Domain-darwin-x64/Example Domain.app","win32metadata":{"ProductName":"","InternalName":"","FileDescription":""}} +1ms
  electron-packager Target Platforms: darwin +0ms
  electron-packager Target Architectures: x64 +0ms
  electron-packager Inferring appVersion from version in /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-16-52-34-app--38131-223qXQkZDU4B/package.json +0ms
  electron-packager Application name: Example Domain +15ms
  electron-packager Target Electron version: 12.0.10 +0ms
  electron-packager Ignored path regular expressions: [
  '/package-lock\\.json$',
  '/yarn\\.lock$',
  '/\\.git($|/)',
  '/node_modules/\\.bin($|/)',
  '\\.o(bj)?$'
] +0ms
  electron-packager Downloading Electron with options {"platform":"darwin","arch":"x64","version":"12.0.10","artifactName":"electron"} +0ms
Packaging app for platform darwin x64 using electron v12.0.10
  electron-packager Creating /Users/USER/darwin-x64-template +5ms
  electron-packager Extracting /Users/USER/Library/Caches/electron/fe82eb0272a083ac9c8100be1631e54e7199d39ec5441c611bd374f326bddbaa/electron-v12.0.10-darwin-x64.zip to /Users/USER/darwin-x64-template +1ms
  electron-packager Initializing app in /Users/USER/Example Domain-darwin-x64 from /Users/USER/darwin-x64-template template +0ms
  electron-packager Ignored paths based on the out param: [
  '/Users/USER/Example Domain-darwin-x64',
  '/Users/USER/Example Domain-darwin-arm64',
  '/Users/USER/Example Domain-linux-ia32',
  '/Users/USER/Example Domain-linux-x64',
  '/Users/USER/Example Domain-linux-armv7l',
  '/Users/USER/Example Domain-linux-arm64',
  '/Users/USER/Example Domain-linux-mips64el',
  '/Users/USER/Example Domain-mas-x64',
  '/Users/USER/Example Domain-mas-arm64',
  '/Users/USER/Example Domain-win32-ia32',
  '/Users/USER/Example Domain-win32-x64',
  '/Users/USER/Example Domain-win32-arm64'
] +4s
  electron-packager Copying icon "/Users/USER/Applications/Example Domain-darwin-x64/Example Domain.app/Contents/Resources/electron.icns" to app's Resources as "electron.icns" +0ms
  electron-packager Renaming Electron to Example Domain in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/MacOS +92ms
  electron-packager Renaming Electron Helper to Example Domain Helper in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper.app/Contents/MacOS +1ms
  electron-packager Renaming Electron Helper (Renderer) to Example Domain Helper (Renderer) in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (Renderer).app/Contents/MacOS +1ms
  electron-packager Renaming Electron Helper (Plugin) to Example Domain Helper (Plugin) in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (Plugin).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (GPU) to Example Domain Helper (GPU) in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (GPU).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper.app to Example Domain Helper.app in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (Renderer).app to Example Domain Helper (Renderer).app in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (Plugin).app to Example Domain Helper (Plugin).app in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/Frameworks +1ms
  electron-packager Renaming Electron Helper (GPU).app to Example Domain Helper (GPU).app in /Users/USER/Example Domain-darwin-x64/Electron.app/Contents/Frameworks +0ms

Finalizing build...
App built to /Users/USER/Example Domain-darwin-x64, move to wherever it makes sense for you and run the app bundle.
```

**Expected behavior**

Saves updated app to ~/Applications

**Actual behavior**
Saves updated app to home folder ~/
(current working directory)

**Debug info**

- Console logs of your `nativefier` build command, with `--verbose` flag
- If the bug happens at app run time, the in-app DevTools console logs (open it with F12)
- Error messages
- Screenshots
- Anything else relevant!

**Context**
Where I discovered this: 
https://github.com/mattruzzi/nativefier-gui/tree/feat--Upgrade-existing-apps
I was trying to add a menu to nativefier-gui update existing nativefier apps and was running the nativfier programmatic API and got the following:
```
Preparing Electron app...

Converting icons...

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn't cached yet...
Packaging app for platform darwin x64 using electron v12.0.10
Error: Invalid package /Users/USER/Documents/github2/nativefier-gui-2/darwin-x64-template/Electron.app/Contents/Resources/default_app.asar
    at createError (electron/js2c/asar_bundle.js:5:1585)
    at Object.t.<computed> [as open] (electron/js2c/asar_bundle.js:5:2419)
    at _openWriteFs (internal/fs/streams.js:370:15)
    at new WriteStream (internal/fs/streams.js:343:5)
    at createWriteStream (fs.js:2008:10)
    at Extractor.extractEntry (/Users/USER/Documents/github2/nativefier-gui-2/node_modules/electron-packager/node_modules/extract-zip/index.js:132:34)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at ZipFile.<anonymous> (/Users/USER/Documents/github2/nativefier-gui-2/node_modules/electron-packager/node_modules/extract-zip/index.js:65:11)
```
```node
function updateApp(path) {
    const appPath = dialog.showOpenDialogSync(null, {
        properties: ["openFile", "openDirectory"],
        buttonLabel: "Update Nativefier App",
    })[0];
    console.log("Upgrading " + appPath);
    const args = { "upgrade": appPath }
    nativefier(args, function (error, appPath) {
        if (error) {
            console.error(error);
            return;
        }
        console.log("App has been nativefied to", appPath);
    });
}
```

 - Nativefier: 44.0.0
 - Node.js: v16.3.0
 - Npm: 7.16.0
 - OS: Mac OS X 10.13.6
 - Is it a regression? If yes, what's the last working / first broken version?
 - Additional context: (for example: "I'm behind a proxy, with configuration X and protocol Y")

### PR DESCRIPTION
**Pull request by TheCleric**
This should resolve #1216 

Found a few bugs and issues with how we were overwriting the old app when upgrading. This led to a change where I've stripped ncp from our actual code (while keeping it for cli use) and replacing it with fs-extra (which I know was some of what was attempted in #1261) . I did this because the ncp calls were not performing correctly in my use case (and possibly in other use cases), and that package is unmaintained.

I also found a bug with global shortcuts not getting passed over when upgrading, so that was resolved as well including testing.

### CODE REVIEW

File: src/integration-test.ts [hunk 1]
```diff
@@ -93,6 +89,21 @@ async function checkApp(
 
   // Test lang
   expect(nativefierConfig?.lang).toEqual(inputOptions.lang);
+
+  // Test global shortcuts
```
Comment (ronjouch) [inline src/integration-test.ts:93]: Nice

**Review by ronjouch — APPROVED** [review]
👍. Just needs a `relock` to fix the shrinkwrap conflict; letting you do it and merge.

**Comment by TheCleric** [comment]
@ronjouch I was having a BEAR of a time getting `npm run relock` to work under windows in its current iteration. My modification seems to be equivalent from my perspective, but can you confirm before merging?

File: package.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
     "lint": "eslint shared app src --ext .ts",
     "list-outdated-deps": "npm out; cd app && npm out; true",
     "prepare": "cd app && npm ci && cd .. && npm run build",
-    "relock": "rm -rf ./node_modules/ ./app/node_modules/ ./npm-shrinkwrap.json ./app/npm-shrinkwrap.json; npm install --ignore-scripts --package-lock && mv package-lock.json npm-shrinkwrap.json && npm out; cd app && npm install --ignore-scripts --package-lock && mv package-lock.json npm-shrinkwrap.json && npm out; cd .. && true",
+    "relock": "rm -rf ./node_modules/ ./app/node_modules/ ./npm-shrinkwrap.json ./app/npm-shrinkwrap.json && npm install --ignore-scripts --package-lock && mv package-lock.json npm-shrinkwrap.json && npm out && cd app && npm install --ignore-scripts --package-lock && mv package-lock.json npm-shrinkwrap.json && npm out",
```
Comment (ronjouch) [inline package.json:46]: @TheCleric hmmm, I understand `true` doesn't exist under Windows CMD, but I thought `;` does and has the same behavior as Unix's `;`, that is to keep chaining regardless of error code. Said differently, doesn't this work?

```suggestion
    "relock": "rm -rf ./node_modules/ ./app/node_modules/ ./npm-shrinkwrap.json ./app/npm-shrinkwrap.json; npm install --ignore-scripts --package-lock && mv package-lock.json npm-shrinkwrap.json && npm out; cd app && npm install --ignore-scripts --package-lock && mv package-lock.json npm-shrinkwrap.json && npm out; cd ..",
```

Comment (TheCleric) [reply package.json:46]: When I get on my window machine I will give that a shot. Maybe it was just the true. Unfortunately all I was really getting was something "that didn't work and it's not our fault" out of npm.

Comment (TheCleric) [reply package.json:46]: Also I tried just using the `npm shrinkwrap` command, but that seems to foul up the `fs-events` stuff in the shrinkwrap, which is the reason I assume you had avoided it.

Comment (ronjouch) [reply package.json:46]: > When I get on my window machine I will give that a shot. Maybe it was just the true. Unfortunately all I was really getting was something "that didn't work and it's not our fault" out of npm.

@TheCleric 👍

> Also I tried just using the `npm shrinkwrap` command, but that seems to foul up the `fs-events` stuff in the shrinkwrap, which is the reason I assume you had avoided it.

@TheCleric yup; if you look at commits from last week you'll see a few commits of me fumbling around `fs-events` nonsense in CI due to differences between `npm install --package-lock`'s and `npm shrinkwrap`. Thus, the current situation :shrug:

**Comment by TheCleric** [comment]
@ronjouch I've added the request for help here.

Screen Shot 2021-10-04 at 1 28 53 PM

**Pull request merged**